README.md
# EyeBreaker
#### _Because your eyes needs it_

In a survey by OnePoll, it was discovered that an average person spends 17 hours and 9 minutes everyday on screen, out of that around 5 hours on a computer/laptop screen time. That\'s a total of 44 years of a person\'s life that is spent watching a display.

This is a serious issue that could lead to variety of eye diseases like Eye fatigue, Dry and irritated eyes, Loss of focus flexibility, Nearsightedness, Retinal damage.
Some of these damages to a person could be permanent, others; very costly to treat. We believe that by a simple set of regular on-screen-off-screen break patterns can help rest the eyes and recover from constant eye screen damages.

That is why, we are introducing EyeBreaker, a simple yet an amazingly helpful tool that will notify you to take breaks at regular interval from your computer/laptop screen so that your eyes can maintain a better long-term health.

The tool uses your device\'s camera/webcam to track your face direction using Machine Learning (Tensorflow) to determine if you are watching the screen or not. If you are not watching the screen or not present in front of the computer, then it considers it as your break period.

### Run this Tool
To run this tool you need to install some python libraries. It would be prefered if you create a new python environment to avoid any version clashes with your system's global libraries.

Python Virtual Environment Reference: https://docs.python.org/3/library/venv.html

Next step is to install all the packages to your environment. After you have sourced your environment with the command:
```
source <venv>/bin/activate
```
You need to install all the libraries that are mentioned in requirements.txt using the command (Pip):
```
pip install -r requirements.txt
```
After all the libraries have been successfully installed you can run your program by running the main.py file:
```
python main.py
```

Then the tool's UI window will successfuly open up (Shown in the screenshots below).
Some of the customization that you can do:
 - Focus Time: How much time you can focus on-screen without taking a break. Default 1200 seconds (20 Minutes).
 - Break Period: Number of seconds you want to take break. Default 60 seconds (1 Minute).
 - Checking Period: At what intervals should the camera check if you are on-screen or off-screen. Default 45 seconds.

When you start using the tool, it will capture snapshots using your camera/webcam at regular interval based on the checking period variable, and will determine if you are watching the screen or taking a break. When you should be taking break after (Focus Time) minutes of screen time, there will be a sound played for you to notify the start and end of the break period.

We have minimized the usage of Memory that this program consumes and with the checking period you can control the usage of how often the camera captures and determines your screentime, thus saving battery consumption.

### UI Screenshots
![Screen Shot 2021-06-27 at 2 49 22 PM](https://user-images.githubusercontent.com/15246084/123539543-ef30a000-d757-11eb-88ac-3c4e8c52f14a.png)
![Screen Shot 2021-06-27 at 2 49 38 PM](https://user-images.githubusercontent.com/15246084/123539548-f35cbd80-d757-11eb-8460-c5d44d2d96df.png)
![Screen Shot 2021-06-27 at 2 49 49 PM](https://user-images.githubusercontent.com/15246084/123539552-f5bf1780-d757-11eb-9d8a-34a68d9c066a.png)


###### Please Note: This application is developed based on open source research and surveys. No Medical Proof has been granted by usage of this application.


Changelog.txt
Todo
- Currently the model only works for when we are looking at the screen straight, train it for other angles as well.


test.py
import time
import sys
import cv2
import numpy as np
import math
import pandas as pd
from face_detector import get_face_detector, find_faces
from face_landmarks import get_landmark_model, detect_marks, draw_marks

face_model = get_face_detector()        # import face model
landmark_model = get_landmark_model()   # import landmark model
cap = cv2.VideoCapture(0)               # record video from webcam, to be accesed via object 'cap'
ret, img = cap.read()                   # read frames out of recorded video (real time)

class_value = int(sys.argv[1])          # ?
timing = int(sys.argv[2])               # Provide Timing (secs?) upto which the frames should be recorded

df_orig = pd.DataFrame()                # a dataframe to store ________
df = pd.DataFrame()                     # a dataframe to store ________

cnt = 0                                 # initialize counter to 0
while True:                             # Infinite loop
    if cnt==timing:                     # If time is up, stop recording frames
        break
    cnt += 1
    ret, img = cap.read()                       # ret == True if frame exists ;  img == frame
    time.sleep(1)                               # a delay of 1 second to induce 1 FPS
    if ret == True:                             # if frame exists
        faces = find_faces(img, face_model)     # find faces from the captured frame
        for face in faces:                      # for each face identified
            orig,marks = detect_marks(img, landmark_model, face)    # detect marks
            
            c = np.reshape(orig,(136,1))        # ??   
            df2 = pd.DataFrame(c)               # ??
            df2 = df2.T                         # ??
            df2['class'] = class_value          # ??
            df_orig = df_orig.append(df2)       # ??

            
            c = np.reshape(marks,(136,1))       # ??
            df2 = pd.DataFrame(c)               # ??
            df2 = df2.T                         # ??
            df2['class'] = class_value          # ??
            df = df.append(df2)                 # ??

            draw_marks(img, marks, color=(0, 255, 0)) # draw the marks in image

        cv2.imshow('img', img)                  # display each captured frame to the user
        if cv2.waitKey(1) == 113:               # ??
            break

df.to_csv('my_csv.csv',mode='a', header=False)                  # save df as CSV
df_orig.to_csv('my_csv_orig.csv',mode='a', header=False)        # save df_orig as CSV


main.py
from trial import process,process_kill
import threading

# The three imports here are non-used : they were hidden and pyinstaller could not identify them 
import scipy
import sklearn
import sklearn.utils._weight_vector
from PyQt5 import QtWidgets
from gui_main import Ui_MainWindow
import sys
# use main.py to also link GUI with the main file

class ApplicationWindow(QtWidgets.QMainWindow):
    def __init__(self):
        super(ApplicationWindow, self).__init__()

        self.ui = Ui_MainWindow()
        self.ui.setupUi(self)
        # self.ui.pushButton_6.clicked.connect(lambda: process(int(self.ui.textEdit.toPlainText()), int(self.ui.textEdit_2.toPlainText()), int(self.ui.textEdit_3.toPlainText())))
        self.ui.pushButton_6.clicked.connect(self.start_process)
        self.ui.pushButton_4.clicked.connect(self.stop_process)

    def stop_process(self):
        process_kill()
        self.x.join()
        print('Stopped')


    def start_process(self):
        try:
            self.x = threading.Thread(target=process, args=(int(self.ui.textEdit.toPlainText()), int(self.ui.textEdit_2.toPlainText()), int(self.ui.textEdit_3.toPlainText())))
            self.x.start()
        except:
            print("Threading failed")



def main():
    app = QtWidgets.QApplication(sys.argv)
    application = ApplicationWindow()
    application.show()
    sys.exit(app.exec_())

if __name__ == '__main__':
    main()

trial.py
import time
import sys
import cv2
import numpy as np
import math
import pandas as pd
from face_detector import get_face_detector, find_faces
from face_landmarks import get_landmark_model, detect_marks, draw_marks
import pickle
import sklearn
from playsound import playsound

playsound('hollow-582.mp3')

face_model = get_face_detector()
landmark_model = get_landmark_model()
loaded_model = pickle.load(open('mymodel.save', 'rb'))
running_process = False

def process_kill():
    global running_process
    running_process = False


def process(focus_time=600,break_time=60,camera_time=45,threshold=7):
    cap = cv2.VideoCapture()
    cap.open(0,cv2.CAP_DSHOW)

    taking_break = True
    buffer = 0
    buffer_2 = 0
    threshold = threshold
    multiple = 1
    start = time.time()
    old_time = start
    global running_process
    running_process = True

    while running_process:
        time.sleep(0.5)
        
        ret, img = cap.read()
        if ret == True:
            faces = find_faces(img, face_model)
            if len(faces) == 1:
                orig,marks = detect_marks(img, landmark_model, faces[0])
                c = np.reshape(marks,(136,1)).T
                res = loaded_model.predict(c)[0]
                print('Straight' if res==1 else 'RELAXED')
                if taking_break:
                    if res==1:
                        buffer+=1
                        if buffer>=threshold:
                            taking_break = False
                            buffer = 0
                            end = time.time()
                            if int(end-start) > break_time:
                                print(f'Break finished after {int(end-start)} seconds.')
                                playsound('hollow-582.mp3')
                                multiple = 1
                                start = time.time()
                            else:
                                start = old_time
                    else:
                        buffer = 0
                else:
                    if res==1:
                        buffer = 0
                        end = time.time()
                        buffer_2 += 1
                        if int(end-start) > focus_time*multiple:
                            multiple += 1
                            print(f'It\'s been {int( (end-start)/60)} minutes. Take a break now!')
                            playsound('hollow-582.mp3')
                        
                        if buffer_2 >3:
                            cap.release()
                            buffer_2 = 0
                            time.sleep(camera_time)
                        
                    else:
                        buffer += 1
                        buffer_2 = 0
                        if buffer>=threshold:
                            print('Taking Break.')
                            taking_break = True
                            old_time = start
                            start = time.time()
            elif len(faces)==0:
                print('Relaxed. No one present.')
                if taking_break:
                    buffer=0
                else:
                    buffer += 1
                    buffer_2 = 0
                    if buffer>=threshold:
                        print('Taking Break.')
                        taking_break = True
                        old_time = start
                        start = time.time()
        else:
            cap.open(0,cv2.CAP_DSHOW)
        
                            
                            

# process()

face_detector.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 29 17:52:00 2020
@author: hp
"""

import cv2
import numpy as np

def get_face_detector(modelFile=None,
                      configFile=None,
                      quantized=False):
    """
    Get the face detection caffe model of OpenCV's DNN module
    
    Parameters
    ----------
    modelFile : string, optional
        Path to model file. The default is "models/res10_300x300_ssd_iter_140000.caffemodel" or models/opencv_face_detector_uint8.pb" based on quantization.
    configFile : string, optional
        Path to config file. The default is "models/deploy.prototxt" or "models/opencv_face_detector.pbtxt" based on quantization.
    quantization: bool, optional
        Determines whether to use quantized tf model or unquantized caffe model. The default is False.
    
    Returns
    -------
    model : dnn_Net

    """
    if quantized:
        if modelFile == None:
            modelFile = "models/opencv_face_detector_uint8.pb"
        if configFile == None:
            configFile = "models/opencv_face_detector.pbtxt"
        model = cv2.dnn.readNetFromTensorflow(modelFile, configFile)
        
    else:
        if modelFile == None:
            modelFile = "models/res10_300x300_ssd_iter_140000.caffemodel"
        if configFile == None:
            configFile = "models/deploy.prototxt"
        model = cv2.dnn.readNetFromCaffe(configFile, modelFile)
    return model

def find_faces(img, model):
    """
    Find the faces in an image
    
    Parameters
    ----------
    img : np.uint8
        Image to find faces from
    model : dnn_Net
        Face detection model

    Returns
    -------
    faces : list
        List of coordinates of the faces detected in the image

    """
    h, w = img.shape[:2]
    blob = cv2.dnn.blobFromImage(cv2.resize(img, (300, 300)), 1.0,
	(300, 300), (104.0, 177.0, 123.0))
    model.setInput(blob)
    res = model.forward()
    faces = []
    for i in range(res.shape[2]):
        confidence = res[0, 0, i, 2]
        if confidence > 0.5:
            box = res[0, 0, i, 3:7] * np.array([w, h, w, h])
            (x, y, x1, y1) = box.astype("int")
            faces.append([x, y, x1, y1])
    return faces

def draw_faces(img, faces):
    """
    Draw faces on image

    Parameters
    ----------
    img : np.uint8
        Image to draw faces on
    faces : List of face coordinates
        Coordinates of faces to draw

    Returns
    -------
    None.

    """
    for x, y, x1, y1 in faces:
        cv2.rectangle(img, (x, y), (x1, y1), (0, 0, 255), 3)
        


face_landmarks.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 29 19:47:08 2020
@author: hp
"""

import cv2
import numpy as np
import tensorflow as tf
from tensorflow import keras


def get_landmark_model(saved_model='models/pose_model'):
    """
    Get the facial landmark model. 
    Original repository: https://github.com/yinguobing/cnn-facial-landmark

    Parameters
    ----------
    saved_model : string, optional
        Path to facial landmarks model. The default is 'models/pose_model'.

    Returns
    -------
    model : Tensorflow model
        Facial landmarks model

    """
    #model = keras.models.load_model(saved_model)
    model = tf.saved_model.load(saved_model)
    return model

def get_square_box(box):
    """Get a square box out of the given box, by expanding it."""
    left_x = box[0]
    top_y = box[1]
    right_x = box[2]
    bottom_y = box[3]

    box_width = right_x - left_x
    box_height = bottom_y - top_y

    # Check if box is already a square. If not, make it a square.
    diff = box_height - box_width
    delta = int(abs(diff) / 2)

    if diff == 0:                   # Already a square.
        return box
    elif diff > 0:                  # Height > width, a slim box.
        left_x -= delta
        right_x += delta
        if diff % 2 == 1:
            right_x += 1
    else:                           # Width > height, a short box.
        top_y -= delta
        bottom_y += delta
        if diff % 2 == 1:
            bottom_y += 1

    # Make sure box is always square.
    assert ((right_x - left_x) == (bottom_y - top_y)), 'Box is not square.'

    return [left_x, top_y, right_x, bottom_y]

def move_box(box, offset):
        """Move the box to direction specified by vector offset"""
        left_x = box[0] + offset[0]
        top_y = box[1] + offset[1]
        right_x = box[2] + offset[0]
        bottom_y = box[3] + offset[1]
        return [left_x, top_y, right_x, bottom_y]

def detect_marks(img, model, face):
    """
    Find the facial landmarks in an image from the faces

    Parameters
    ----------
    img : np.uint8
        The image in which landmarks are to be found
    model : Tensorflow model
        Loaded facial landmark model
    face : list
        Face coordinates (x, y, x1, y1) in which the landmarks are to be found

    Returns
    -------
    marks : numpy array
        facial landmark points

    """

    offset_y = int(abs((face[3] - face[1]) * 0.1))
    box_moved = move_box(face, [0, offset_y])
    facebox = get_square_box(box_moved)
    
    h, w = img.shape[:2]
    if facebox[0] < 0:
        facebox[0] = 0
    if facebox[1] < 0:
        facebox[1] = 0
    if facebox[2] > w:
        facebox[2] = w
    if facebox[3] > h:
        facebox[3] = h
    
    face_img = img[facebox[1]: facebox[3],
                     facebox[0]: facebox[2]]
    face_img = cv2.resize(face_img, (128, 128))
    face_img = cv2.cvtColor(face_img, cv2.COLOR_BGR2RGB)
    
    # # Actual detection.
    predictions = model.signatures["predict"](
        tf.constant([face_img], dtype=tf.uint8))

    # Convert predictions to landmarks.
    marks = np.array(predictions['output']).flatten()[:136]
    marks = np.reshape(marks, (-1, 2))
    orig = np.copy(marks)

    marks *= (facebox[2] - facebox[0])
    marks[:, 0] += facebox[0]
    marks[:, 1] += facebox[1]
    marks = marks.astype(np.uint)

    return orig,marks

def draw_marks(image, marks, color=(0, 255, 0)):
    """
    Draw the facial landmarks on an image

    Parameters
    ----------
    image : np.uint8
        Image on which landmarks are to be drawn.
    marks : list or numpy array
        Facial landmark points
    color : tuple, optional
        Color to which landmarks are to be drawn with. The default is (0, 255, 0).

    Returns
    -------
    None.

    """
    for mark in marks:
        cv2.circle(image, (mark[0], mark[1]), 2, color, -1, cv2.LINE_AA)
    


trainer/trainer.py
"""
    An automated program that opens webcam and records video of the user.
    Recording while pressing the button 'F' should be segregated as video clip while looking at monitor.
    Recording while pressing 'T' should be segregated as video clip looking away from monitor.
    Recorded video are to be split into images.

    xtra : track memory consumption of keeping webcam on
"""
import sys
import pathlib
import cv2
import time
from datetime import datetime

import os
import psutil
process = psutil.Process(os.getpid())

def main():

    option = sys.argv[1]
    timestamp = datetime.now().strftime('%Y-%m-%d %H:%M:%S')

    if option == "facing":
        _path = 'data_collection/facing/'+timestamp
    else:
        _path = 'data_collection/not_facing/'+timestamp
    
    #=================================================================================================================#

    vid = cv2.VideoCapture(0)    
    pathlib.Path(_path).mkdir(exist_ok=True) # Create a folder with timestamp to store images

    #=================================================================================================================#
    sec = 0
    framerate = 1
    count = 1
    while(True):
        frame_exists,frame = vid.read()
        cv2.imshow('frame', frame)  
        print("Usage: " + str(process.memory_info().rss/1000000) + " MB")
        if frame_exists:
            cv2.imwrite(_path+"/%d.jpg"%count,frame)
        time.sleep(1)
        count += 1
        if cv2.waitKey(1) & 0xFF == ord('q'):
            break

    vid.release()
    cv2.destroyAllWindows()

   

main()


gui_main.py
# -*- coding: utf-8 -*-

# Form implementation generated from reading ui file 'main.ui'
#
# Created by: PyQt5 UI code generator 5.9.2
#
# WARNING! All changes made in this file will be lost!

from PyQt5 import QtCore, QtGui, QtWidgets

class Ui_MainWindow(object):
    def setupUi(self, MainWindow):
        MainWindow.setObjectName("MainWindow")
        MainWindow.resize(800, 600)
        self.centralwidget = QtWidgets.QWidget(MainWindow)
        self.centralwidget.setObjectName("centralwidget")
        self.tabWidget = QtWidgets.QTabWidget(self.centralwidget)
        self.tabWidget.setGeometry(QtCore.QRect(20, 10, 751, 551))
        self.tabWidget.setObjectName("tabWidget")
        self.tab_3 = QtWidgets.QWidget()
        self.tab_3.setObjectName("tab_3")
        self.label_11 = QtWidgets.QLabel(self.tab_3)
        self.label_11.setGeometry(QtCore.QRect(240, 40, 261, 71))
        font = QtGui.QFont()
        font.setFamily("Yrsa")
        font.setPointSize(32)
        font.setBold(True)
        font.setItalic(False)
        font.setWeight(75)
        self.label_11.setFont(font)
        self.label_11.setAlignment(QtCore.Qt.AlignCenter)
        self.label_11.setObjectName("label_11")
        self.label_17 = QtWidgets.QLabel(self.tab_3)
        self.label_17.setGeometry(QtCore.QRect(230, 110, 281, 31))
        font = QtGui.QFont()
        font.setFamily("aakar")
        font.setPointSize(14)
        font.setItalic(True)
        self.label_17.setFont(font)
        self.label_17.setAlignment(QtCore.Qt.AlignCenter)
        self.label_17.setObjectName("label_17")
        self.label_2 = QtWidgets.QLabel(self.tab_3)
        self.label_2.setGeometry(QtCore.QRect(20, 160, 701, 341))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.label_2.setFont(font)
        self.label_2.setAlignment(QtCore.Qt.AlignLeading|QtCore.Qt.AlignLeft|QtCore.Qt.AlignTop)
        self.label_2.setWordWrap(True)
        self.label_2.setObjectName("label_2")
        self.tabWidget.addTab(self.tab_3, "")
        self.tab = QtWidgets.QWidget()
        self.tab.setObjectName("tab")
        self.label_12 = QtWidgets.QLabel(self.tab)
        self.label_12.setGeometry(QtCore.QRect(230, 110, 281, 31))
        font = QtGui.QFont()
        font.setFamily("aakar")
        font.setPointSize(14)
        font.setItalic(True)
        self.label_12.setFont(font)
        self.label_12.setAlignment(QtCore.Qt.AlignCenter)
        self.label_12.setObjectName("label_12")
        self.label_10 = QtWidgets.QLabel(self.tab)
        self.label_10.setGeometry(QtCore.QRect(240, 40, 261, 71))
        font = QtGui.QFont()
        font.setFamily("Yrsa")
        font.setPointSize(32)
        font.setBold(True)
        font.setItalic(False)
        font.setWeight(75)
        self.label_10.setFont(font)
        self.label_10.setAlignment(QtCore.Qt.AlignCenter)
        self.label_10.setObjectName("label_10")
        self.pushButton_6 = QtWidgets.QPushButton(self.tab)
        self.pushButton_6.setGeometry(QtCore.QRect(480, 300, 221, 61))
        self.pushButton_6.setObjectName("pushButton_6")
        self.line_3 = QtWidgets.QFrame(self.tab)
        self.line_3.setGeometry(QtCore.QRect(400, 220, 16, 271))
        self.line_3.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_3.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_3.setObjectName("line_3")
        self.label_13 = QtWidgets.QLabel(self.tab)
        self.label_13.setGeometry(QtCore.QRect(50, 230, 291, 31))
        font = QtGui.QFont()
        font.setPointSize(16)
        font.setBold(True)
        font.setWeight(75)
        self.label_13.setFont(font)
        self.label_13.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.label_13.setAlignment(QtCore.Qt.AlignCenter)
        self.label_13.setObjectName("label_13")
        self.label_5 = QtWidgets.QLabel(self.tab)
        self.label_5.setGeometry(QtCore.QRect(40, 340, 181, 31))
        self.label_5.setObjectName("label_5")
        self.textEdit = QtWidgets.QTextEdit(self.tab)
        self.textEdit.setGeometry(QtCore.QRect(230, 300, 161, 31))
        self.textEdit.setInputMethodHints(QtCore.Qt.ImhMultiLine)
        self.textEdit.setObjectName("textEdit")
        self.textEdit_2 = QtWidgets.QTextEdit(self.tab)
        self.textEdit_2.setGeometry(QtCore.QRect(230, 340, 161, 31))
        self.textEdit_2.setObjectName("textEdit_2")
        self.label_8 = QtWidgets.QLabel(self.tab)
        self.label_8.setGeometry(QtCore.QRect(40, 300, 181, 31))
        self.label_8.setObjectName("label_8")
        self.label_6 = QtWidgets.QLabel(self.tab)
        self.label_6.setGeometry(QtCore.QRect(40, 380, 181, 31))
        self.label_6.setObjectName("label_6")
        self.textEdit_3 = QtWidgets.QTextEdit(self.tab)
        self.textEdit_3.setGeometry(QtCore.QRect(230, 380, 161, 31))
        self.textEdit_3.setObjectName("textEdit_3")
        self.label_16 = QtWidgets.QLabel(self.tab)
        self.label_16.setGeometry(QtCore.QRect(50, 260, 291, 31))
        font = QtGui.QFont()
        font.setPointSize(12)
        self.label_16.setFont(font)
        self.label_16.setCursor(QtGui.QCursor(QtCore.Qt.ArrowCursor))
        self.label_16.setAlignment(QtCore.Qt.AlignCenter)
        self.label_16.setObjectName("label_16")
        self.tabWidget.addTab(self.tab, "")
        self.tab_2 = QtWidgets.QWidget()
        self.tab_2.setObjectName("tab_2")
        self.pushButton_4 = QtWidgets.QPushButton(self.tab_2)
        self.pushButton_4.setGeometry(QtCore.QRect(480, 300, 221, 61))
        self.pushButton_4.setObjectName("pushButton_4")
        self.line_2 = QtWidgets.QFrame(self.tab_2)
        self.line_2.setGeometry(QtCore.QRect(400, 220, 16, 271))
        self.line_2.setFrameShape(QtWidgets.QFrame.VLine)
        self.line_2.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line_2.setObjectName("line_2")
        self.label_14 = QtWidgets.QLabel(self.tab_2)
        self.label_14.setGeometry(QtCore.QRect(240, 40, 261, 71))
        font = QtGui.QFont()
        font.setFamily("Yrsa")
        font.setPointSize(32)
        font.setBold(True)
        font.setItalic(False)
        font.setWeight(75)
        self.label_14.setFont(font)
        self.label_14.setAlignment(QtCore.Qt.AlignCenter)
        self.label_14.setObjectName("label_14")
        self.label_15 = QtWidgets.QLabel(self.tab_2)
        self.label_15.setGeometry(QtCore.QRect(230, 110, 281, 31))
        font = QtGui.QFont()
        font.setFamily("aakar")
        font.setPointSize(14)
        font.setItalic(True)
        self.label_15.setFont(font)
        self.label_15.setAlignment(QtCore.Qt.AlignCenter)
        self.label_15.setObjectName("label_15")
        self.label = QtWidgets.QLabel(self.tab_2)
        self.label.setGeometry(QtCore.QRect(60, 240, 291, 171))
        self.label.setLineWidth(1)
        self.label.setMidLineWidth(0)
        self.label.setScaledContents(False)
        self.label.setAlignment(QtCore.Qt.AlignCenter)
        self.label.setWordWrap(True)
        self.label.setObjectName("label")
        self.tabWidget.addTab(self.tab_2, "")
        MainWindow.setCentralWidget(self.centralwidget)
        self.statusbar = QtWidgets.QStatusBar(MainWindow)
        self.statusbar.setObjectName("statusbar")
        MainWindow.setStatusBar(self.statusbar)

        self.retranslateUi(MainWindow)
        self.tabWidget.setCurrentIndex(1)
        QtCore.QMetaObject.connectSlotsByName(MainWindow)

    def retranslateUi(self, MainWindow):
        _translate = QtCore.QCoreApplication.translate
        MainWindow.setWindowTitle(_translate("MainWindow", "MainWindow"))
        self.label_11.setText(_translate("MainWindow", "EyeBreaker"))
        self.label_17.setText(_translate("MainWindow", "\"Beause your eyes needs it\""))
        self.label_2.setText(_translate("MainWindow", "In a survey by OnePoll, it was discovered that an average person spends 17 hours and 9 minutes everyday on screen, out of that around 5 hours on a computer/laptop screen time. That\'s a total of 44 years of a person\'s life that is spent watching a display. \n"
"\n"
"This is a serious issue that could lead to variety of eye diseases like Eye fatigue, Dry and irritated eyes, Loss of focus flexibility, Nearsightedness, Retinal damage. \n"
"Some of these damages to a person could be permanent, others; very costly to treat. We believe that by a simple set of regular on-screen-off-screen break patterns can help rest the eyes and recover from constant eye screen damages.\n"
"\n"
"That is why, we are introducing EyeBreaker, a simple yet an amazingly helpful tool that will notify you to take breaks at regular interval from your computer/laptop screen so that your eyes can maintain a better long-term health. \n"
"The tool uses your device\'s camera/webcam to track your face direction to determine if you are watching the screen or not. If you are not watching the screen or not present in front of the computer, then it considers it as your break period. \n"
"\n"
"Some of the customization that you can do:\n"
" - Focus Time: How much time you can focus on-screen without taking a break.\n"
" - Break Period: Number of seconds you want to take break.\n"
" - Checking Period: At what intervals should the camera check if you are on-screen or off-screen.\n"
"\n"
"Please Note: This application is developed based on open source research and surveys. No Medical Proof has been granted by usage of this application."))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab_3), _translate("MainWindow", "Info"))
        self.label_12.setText(_translate("MainWindow", "\"Beause your eyes needs it\""))
        self.label_10.setText(_translate("MainWindow", "EyeBreaker"))
        self.pushButton_6.setText(_translate("MainWindow", "Run"))
        self.label_13.setText(_translate("MainWindow", "Customize"))
        self.label_5.setText(_translate("MainWindow", "- Break Period : "))
        self.textEdit.setHtml(_translate("MainWindow", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'.SF NS Text\'; font-size:13pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">1200</p></body></html>"))
        self.textEdit.setPlaceholderText(_translate("MainWindow", "1200"))
        self.textEdit_2.setHtml(_translate("MainWindow", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'.SF NS Text\'; font-size:13pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">60</p></body></html>"))
        self.textEdit_2.setPlaceholderText(_translate("MainWindow", "60"))
        self.label_8.setText(_translate("MainWindow", "- Focus Time : "))
        self.label_6.setText(_translate("MainWindow", "- Checking Period : "))
        self.textEdit_3.setHtml(_translate("MainWindow", "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.0//EN\" \"http://www.w3.org/TR/REC-html40/strict.dtd\">\n"
"<html><head><meta name=\"qrichtext\" content=\"1\" /><style type=\"text/css\">\n"
"p, li { white-space: pre-wrap; }\n"
"</style></head><body style=\" font-family:\'.SF NS Text\'; font-size:13pt; font-weight:400; font-style:normal;\">\n"
"<p style=\" margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;\">45</p></body></html>"))
        self.textEdit_3.setPlaceholderText(_translate("MainWindow", "45"))
        self.label_16.setText(_translate("MainWindow", "(All in seconds)"))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab), _translate("MainWindow", "Start"))
        self.pushButton_4.setText(_translate("MainWindow", "Stop"))
        self.label_14.setText(_translate("MainWindow", "EyeBreaker"))
        self.label_15.setText(_translate("MainWindow", "\"Beause your eyes needs it\""))
        self.label.setText(_translate("MainWindow", "When you are done with your On-Screen work, you can stop EyeBreaker."))
        self.tabWidget.setTabText(self.tabWidget.indexOf(self.tab_2), _translate("MainWindow", "Stop"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    MainWindow = QtWidgets.QMainWindow()
    ui = Ui_MainWindow()
    ui.setupUi(MainWindow)
    MainWindow.show()
    sys.exit(app.exec_())

